frame/support/test/tests/construct_runtime.rs

```diff
@@ -112,8 +112,53 @@ mod module2 {
 	}
 }
 
+mod nested {
+	use super::*;
+
+	pub mod module3 {
+		use super::*;
+
+		pub trait Config: system::Config {}
+
+		frame_support::decl_module! {
+			pub struct Module<T: Config> for enum Call
+				where origin: <T as system::Config>::Origin, system=system
+			{
+				#[weight = 0]
+				pub fn fail(_origin) -> frame_support::dispatch::DispatchResult {
+					Err(Error::<T>::Something.into())
+				}
+
+				fn integrity_test() {
+					INTEGRITY_TEST_EXEC.with(|i| *i.borrow_mut() += 1);
+				}
+			}
+		}
+
+		#[derive(Clone, PartialEq, Eq, Debug, codec::Encode, codec::Decode)]
+		pub struct Origin;
+
+		frame_support::decl_event! {
+			pub enum Event {
+				A,
+			}
+		}
+
+		frame_support::decl_error! {
+			pub enum Error for Module<T: Config> {
+				Something
+			}
+		}
+
+		frame_support::decl_storage! {
+			trait Store for Module<T: Config> as Module {}
+		}
```

Comment:
Huh... how do you specify that I'm not using the default `frame_support::Config` as a supertrait for my Config, and therefore all calls will have an origin that is a type from `system::Config::Origin` rather than `frame_support::Config::Origin`?

Reply:
Ah sorry I didn't the this test relies on this specific system module. you are right